plugin-server/src/utils/db/db.ts

```diff
@@ -1902,12 +1902,43 @@ export class DB {
         })
     }
 
-    public async getPluginSource(pluginId: Plugin['id'], filename = 'index.ts'): Promise<string | null> {
+    public async getPluginSource(pluginId: Plugin['id'], filename: string): Promise<string | null> {
         const { rows }: { rows: { source: string }[] } = await this.postgresQuery(
             `SELECT source FROM posthog_pluginsourcefile WHERE plugin_id = $1 AND filename = $2`,
             [pluginId, filename],
             'getPluginSource'
         )
         return rows[0]?.source
     }
+
+    public async setPluginTranspiled(pluginId: Plugin['id'], filename: string, transpiled: string): Promise<void> {
+        await this.postgresQuery(
+            `INSERT INTO posthog_pluginsourcefile (id, plugin_id, filename, status, transpiled) VALUES($1, $2, $3, $4, $5)
+                ON CONFLICT ON CONSTRAINT unique_filename_for_plugin
+                DO UPDATE SET status = $4, transpiled = $5, error = NULL`,
+            [new UUIDT().toString(), pluginId, filename, 'TRANSPILED', transpiled],
```

Comment:
This this is an enum in Python, this might be a bit easier for other devs to understand if the plugin server had a `PluginSourceFileStatus` enum too.